### PR TITLE
Add new package to evaluate against a number of different datasets

### PR DESCRIPTION
**Pull request by glpuga**
### Proposed changes

_Describe the big picture of your changes here to communicate to the maintainers why we should accept this pull request.
If it fixes a bug or resolves a feature request, be sure to link to that issue._

#### Type of change

- [ ] 🐛 Bugfix (change which fixes an issue)
- [x] 🚀 Feature (change which adds functionality)
- [ ] 📚 Documentation (change which fixes or extends documentation)

💥 **Breaking change!** _Explain why a non-backwards compatible change is necessary or remove this line entirely if not applicable._

### Checklist

_Put an `x` in the boxes that apply. This is simply a reminder of what we will require before merging your code._

- [ ] Lint and unit tests (if any) pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)

### Additional comments

_Anything worth mentioning to the reviewers._

### CODE REVIEW

File: src/benchmarks/beluga_vs_nav2_multi_dataset/reports/nominal_report/sections/datasets/diff_drive_sim_24hs_description.inc [hunk 1]
```diff
@@ -0,0 +1,37 @@
+Dataset information
+^^^^^^^^^^^^^^^^^^^
+
+This is a synthetic dataset created recording a simulated robot wandering around a
+450m^2 office environment for 24 hours.
```
Comment (hidmic) [inline src/benchmarks/beluga_vs_nav2_multi_dataset/reports/nominal_report/sections/datasets/diff_drive_sim_24hs_description.inc:5]: @glpuga nit: consider using math expressions for this:
```suggestion
:math:`450 \mathrm{m}^2` office environment for 24 hours.
```

File: src/benchmarks/beluga_vs_nav2_multi_dataset/reports/nominal_report/sections/datasets/diff_drive_sim_24hs_description.inc [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+
+The simulated diff-drive robot is modelled after a Kobuki platform with a mounted RPLidar A1 (range: 8m). The
```
Comment (hidmic) [inline src/benchmarks/beluga_vs_nav2_multi_dataset/reports/nominal_report/sections/datasets/diff_drive_sim_24hs_description.inc:7]: @glpuga nit:
```suggestion
The simulated diff-drive robot is modelled after a Kobuki platform featuring an RPLidar A1, an 8m range 2D lidar. The
```
Also, links to Kobuki and RPLidar sites would be nice.

File: src/benchmarks/beluga_vs_nav2_multi_dataset/reports/nominal_report/sections/datasets/magazino_datasets_description.inc [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+
+Transformations to the original dataset
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+To repurpose the TorWIC Mapping dataset for 2D localization evaluation, we made the following changes from
+the original dataset as distributed in here:
+
+- The original ROS 1 bagfiles were converted to ROS 2 bagfiles.
+- All topics not related to lidar-based 2D localization were removed to reduce file size.
+- A reference ground-truth and a matching occupancy map were generated for each bagfile using SLAM.
```
Comment (hidmic) [inline src/benchmarks/beluga_vs_nav2_multi_dataset/reports/nominal_report/sections/datasets/magazino_datasets_description.inc:31]: @glpuga does all this apply? This isn't the TorWIC dataset.

File: src/benchmarks/beluga_vs_nav2_multi_dataset/reports/nominal_report/sections/datasets/omni_drive_sim_24hs_description.inc [hunk 4]
```diff
@@ -0,0 +1,40 @@
+Dataset information
+^^^^^^^^^^^^^^^^^^^
+
+This is a synthetic dataset created recording a simulated omni-drive robot wandering around the
+`AWS Robomaker Bookstore World <https://github.com/aws-robotics/aws-robomaker-bookstore-world>`_ for 24 hours.
+
+The simulated omni-drive robot is modelled after a customized Robomaster EP with a mounted RPLidar A2 (range: 12m).
```
Comment (hidmic) [inline src/benchmarks/beluga_vs_nav2_multi_dataset/reports/nominal_report/sections/datasets/omni_drive_sim_24hs_description.inc:7]: @glpuga nit: 
```suggestion
The simulated omni-drive robot is modelled after a customized Robomaster EP featuring an RPLidar A2, a 12m range 2D lidar.
```
Also, links.

File: src/benchmarks/beluga_vs_nav2_multi_dataset/reports/nominal_report/sections/datasets/omni_drive_sim_24hs_description.inc [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+The dataset was recorded in a single run, and it consists of a single continuous bagfile with a duration of 24 hours.
+During this time the robot is periodically sent navigation goals to reachable locations within the map to ensure it
+keeps moving.
+
+The robot covers approximately 45 km during the 24 hours, with an average speed of about 0.5 m/s.
+
+The robot model was configured to introduce small imperfections in the odometry to cause a small amount of drift and
+emulate conditions closer to real-world.
+
+Most, but not all, of the obstacles and furniture are mapped, and one of the walls is windowed and therefore
```
Comment (hidmic) [inline src/benchmarks/beluga_vs_nav2_multi_dataset/reports/nominal_report/sections/datasets/omni_drive_sim_24hs_description.inc:19]: @glpuga 
```suggestion
Most but not all obstacles and furniture are mapped, and one of the walls is windowed and therefore
```

File: src/benchmarks/beluga_vs_nav2_multi_dataset/reports/nominal_report/sections/datasets/openloris_office_description.inc [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+
+Transformations to the original dataset
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+To repurpose the TorWIC Mapping dataset for 2D localization evaluation, we made the following changes from
+the original dataset as distributed in here:
+
+- The original ROS 1 bagfiles were converted to ROS 2 bagfiles.
+- All topics not related to lidar-based 2D localization were removed to reduce file size.
+- A reference ground-truth and a matching occupancy map were generated for each bagfile using SLAM.
```
Comment (hidmic) [inline src/benchmarks/beluga_vs_nav2_multi_dataset/reports/nominal_report/sections/datasets/openloris_office_description.inc:47]: @glpuga same question as in the Magazino datasets. Looks like an unintentional duplicate.

File: src/benchmarks/beluga_vs_nav2_multi_dataset/reports/nominal_report/sections/datasets/torwic_slam_description.inc [hunk 7]
```diff
@@ -0,0 +1,69 @@
...
+
+.. figure:: assets/representative_torwic_slam_map.png
+   :scale: 50 %
+
+   Localization map of one of the TorWIC SLAM dataset trajectories, representative of the environment.
+
+Transformations to the original dataset
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+To repurpose the TorWIC Mapping dataset for 2D localization evaluation, we made the following changes from
```
Comment (hidmic) [inline src/benchmarks/beluga_vs_nav2_multi_dataset/reports/nominal_report/sections/datasets/torwic_slam_description.inc:61]: @glpuga TorWIC SLAM? Better review the transformation listing too.

File: src/benchmarks/beluga_vs_nav2_multi_dataset/reports/nominal_report/sections/datasets/willow_garage_description.inc [hunk 8]
```diff
@@ -0,0 +1,56 @@
...
+
+.. figure:: assets/representative_willow_garage_map.png
+   :scale: 99 %
+
+   Localization map of one of the trajectories in the Willow Garage dataset, representative of the environment.
+
+Transformations to the original dataset
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+To repurpose the TorWIC Mapping dataset for 2D localization evaluation, we made the following changes from
```
Comment (hidmic) [inline src/benchmarks/beluga_vs_nav2_multi_dataset/reports/nominal_report/sections/datasets/willow_garage_description.inc:39]: @glpuga same comments as in elsewhere.

File: src/benchmarks/beluga_vs_nav2_multi_dataset/reports/nominal_report/sections/main_corpus.inc.bkp [hunk 9]
```diff
@@ -0,0 +1,237 @@
+
```
Comment (hidmic) [inline src/benchmarks/beluga_vs_nav2_multi_dataset/reports/nominal_report/sections/main_corpus.inc.bkp:1]: @glpuga a `vim` backup?

File: src/benchmarks/beluga_vs_nav2_multi_dataset/reports/nominal_report/sections/platform.inc [hunk 10]
```diff
@@ -0,0 +1,132 @@
+Methodology
+-----------
+
+Datasets used
+^^^^^^^^^^^^^
+
+A number of pre-recorded datasets were selected to evaluate the performance of the Beluga AMCL and Nav2 AMCL localization systems.
+A number of public datasets were considered for this evaluation, looking for a diversity of environments and robots that were
+that could be considered representative of a wide range of real-world scenarios.
```
Comment (hidmic) [inline src/benchmarks/beluga_vs_nav2_multi_dataset/reports/nominal_report/sections/platform.inc:9]: @glpuga 
```suggestion
A number of public datasets were considered for this evaluation, looking for a diversity of environments and robots that were representative of a wide range of real-world scenarios.
```

File: src/benchmarks/beluga_vs_nav2_multi_dataset/scripts/nominal.robot [hunk 11]
```diff
@@ -0,0 +1,210 @@
...
+    Sets map_filename launch argument to ${artifacts_path}/map.yaml
+    Sets laser_model_type launch argument to ${laser_model}
+    Sets global_frame_id launch argument to ${map_frame}
+    Sets odom_frame_id launch argument to ${odom_frame}
+    Sets base_frame_id launch argument to ${base_frame}
+    Sets scan_topic launch argument to ${scan_topic}
+    Sets initial_pose_x launch argument to ${initial_pose_x}
+    Sets initial_pose_y launch argument to ${initial_pose_y}
+    Sets initial_pose_yaw launch argument to ${initial_pose_yaw}
+    Sets robot_model_type launch argument to ${robot_model_type}
```
Comment (hidmic) [inline src/benchmarks/beluga_vs_nav2_multi_dataset/scripts/nominal.robot:201]: @glpuga meta: thinking out loud, LAMBKIN could use a mechanism for forward entire collections of arguments to an underlying launch file, automagically.

File: src/benchmarks/beluga_vs_nav2_multi_dataset/reports/nominal_report/sections/main_corpus.inc [hunk 12]
```diff
@@ -0,0 +1,167 @@
...
+        'TorWIC SLAM',
+        'Willow Garage',
+        'Diff Drive Sim 24hs',
+        'Omni Drive Sim 24hs',
+    ]
+    datasets_in_data = ape.index.get_level_values(0).unique()
+    missing_datasets = [x for x in datasets_in_data if x not in desired_section_order]
+    desired_section_order = desired_section_order + missing_datasets
+
+    for case_name in desired_section_order:
```
Comment (hidmic) [inline src/benchmarks/beluga_vs_nav2_multi_dataset/reports/nominal_report/sections/main_corpus.inc:85]: @glpuga meta: I wonder if `.. datatemplate:import-module:: lambkin.data` wouldn't be a slightly more sane approach to templating (see https://sphinxcontribdatatemplates.readthedocs.io/en/latest).

File: src/benchmarks/beluga_vs_nav2_multi_dataset/reports/nominal_report/sections/main_corpus.inc [hunk 13]
```diff
@@ -0,0 +1,167 @@
...
+                bagpath = bagfile_location_data.loc[case_name].loc[bagfile].loc['basedir']
+                metadata = get_bagfile_info(os.path.join('..', '..', bagpath, 'bagfiles', bagfile))
+                metadata = [x for x in metadata.split('\n') if "Topic:" not in x]
+                metadata = "    " + "\n    ".join(metadata)
+                mapfile = os.path.join('..', '..', bagpath, 'artifacts', bagfile, 'map.pgm')
+            except:
+                pass
+            output.append(BAGFILE_SECTION_TEMPLATE.format(bagfile_name=bagfile, metadata=metadata))
+            data_for_bagfile = data_for_case.loc[bagfile]
+            try:
```
Comment (hidmic) [inline src/benchmarks/beluga_vs_nav2_multi_dataset/reports/nominal_report/sections/main_corpus.inc:103]: @glpuga why so many `try..except` blocks?

**Review by hidmic — COMMENTED** [review]
First pass, awesome work @glpuga !

File: src/benchmarks/beluga_vs_nav2_multi_dataset/scripts/nominal.robot (hunk 11 above)
Comment (glpuga) [reply src/benchmarks/beluga_vs_nav2_multi_dataset/scripts/nominal.robot:201]: As long as its just less verbose, but not less explicit, that's ok.

My personal point of view is that the more automagic things this does, the harder it is to make sense of how it works.

File: src/benchmarks/beluga_vs_nav2_multi_dataset/reports/nominal_report/sections/main_corpus.inc (hunk 13 above)
Comment (glpuga) [reply src/benchmarks/beluga_vs_nav2_multi_dataset/reports/nominal_report/sections/main_corpus.inc:103]: Becase lambkin frequently fails a test case, and yet moves on leaving a hole in the data. Without this, even a single instance of that would prevent me from creating the output pdf. I use this to try to power-through and just display "???" in the tables that have missing data.

File: src/benchmarks/beluga_vs_nav2_multi_dataset/reports/nominal_report/sections/main_corpus.inc.bkp (hunk 9 above)
Comment (glpuga) [reply src/benchmarks/beluga_vs_nav2_multi_dataset/reports/nominal_report/sections/main_corpus.inc.bkp:1]: nop, I'm not a VIM user.

This is just a manual back up of the code I had to remove to be able to generate the pdf skipping the failing timem data. Loading the timem data fails in the very first loading of the data, so the try-excepts were not enough to get through this failure mode.

This needs to be restored once I figure why timem data is corrupted.

File: src/benchmarks/beluga_vs_nav2_multi_dataset/scripts/nominal.robot (hunk 11 above)
Comment (hidmic) [reply src/benchmarks/beluga_vs_nav2_multi_dataset/scripts/nominal.robot:201]: >  the more automagic things this does, the harder it is to make sense of how it works.

That can be true, yes. In this case, perhaps a macro to invoke forwarding would at least require the user to explicitly request this. Logs may also help (and we are a bit loose in that sense in general, the troubleshooting experience is not great right now). Anyways, just a thought, nothing to do here.

File: src/benchmarks/beluga_vs_nav2_multi_dataset/reports/nominal_report/sections/main_corpus.inc (hunk 13 above)
Comment (hidmic) [reply src/benchmarks/beluga_vs_nav2_multi_dataset/reports/nominal_report/sections/main_corpus.inc:103]: Hmm, doesn't test case failure stop the entire benchmark? :thinking: 

About holes in data. Maybe backfilling the dataset with NaNs or "" could be a useful idiom to make report generation robust to flakey benchmarks. We don't really support that use case well.

Comment (glpuga) [reply src/benchmarks/beluga_vs_nav2_multi_dataset/reports/nominal_report/sections/main_corpus.inc:103]: > Hmm, doesn't test case failure stop the entire benchmark? 🤔

Most of the time it does, but for some reason not always.

File: src/benchmarks/beluga_vs_nav2_multi_dataset/reports/nominal_report/sections/main_corpus.inc.bkp (hunk 9 above)
Comment (glpuga) [reply src/benchmarks/beluga_vs_nav2_multi_dataset/reports/nominal_report/sections/main_corpus.inc.bkp]: done.

File: src/benchmarks/beluga_vs_nav2_multi_dataset/reports/nominal_report/sections/datasets/magazino_datasets_description.inc (hunk 3 above)
Comment (glpuga) [reply src/benchmarks/beluga_vs_nav2_multi_dataset/reports/nominal_report/sections/datasets/magazino_datasets_description.inc:31]: Fixed.

File: src/benchmarks/beluga_vs_nav2_multi_dataset/reports/nominal_report/sections/datasets/openloris_office_description.inc (hunk 6 above)
Comment (glpuga) [reply src/benchmarks/beluga_vs_nav2_multi_dataset/reports/nominal_report/sections/datasets/openloris_office_description.inc:47]: Fixed.

File: src/benchmarks/beluga_vs_nav2_multi_dataset/reports/nominal_report/sections/datasets/torwic_slam_description.inc (hunk 7 above)
Comment (glpuga) [reply src/benchmarks/beluga_vs_nav2_multi_dataset/reports/nominal_report/sections/datasets/torwic_slam_description.inc:61]: Done.

File: src/benchmarks/beluga_vs_nav2_multi_dataset/reports/nominal_report/sections/datasets/willow_garage_description.inc (hunk 8 above)
Comment (glpuga) [reply src/benchmarks/beluga_vs_nav2_multi_dataset/reports/nominal_report/sections/datasets/willow_garage_description.inc:39]: Fixed.

File: src/benchmarks/beluga_vs_nav2_multi_dataset/reports/nominal_report/sections/datasets/diff_drive_sim_24hs_description.inc (hunk 2 above)
Comment (glpuga) [reply src/benchmarks/beluga_vs_nav2_multi_dataset/reports/nominal_report/sections/datasets/diff_drive_sim_24hs_description.inc]: Added links.

File: src/benchmarks/beluga_vs_nav2_multi_dataset/reports/nominal_report/sections/datasets/omni_drive_sim_24hs_description.inc (hunk 4 above)
Comment (glpuga) [reply src/benchmarks/beluga_vs_nav2_multi_dataset/reports/nominal_report/sections/datasets/omni_drive_sim_24hs_description.inc]: Added links.

File: src/benchmarks/beluga_vs_nav2_multi_dataset/reports/nominal_report/sections/main_corpus.inc (hunk 13 above)
Comment (glpuga) [reply src/benchmarks/beluga_vs_nav2_multi_dataset/reports/nominal_report/sections/main_corpus.inc:154]: Notice that LAMBKIN **not stopping** is the preferred behavior.

LAMBKIN should be fault-tolerant, and continue to do as much work as possible, while tagging bad test cases, and allowing to rerun those that failed.

It's no good if a four-day run fails on Saturday at 6am, and the user figures on Monday.

Comment (hidmic) [reply src/benchmarks/beluga_vs_nav2_multi_dataset/reports/nominal_report/sections/main_corpus.inc:154]: Hmm, that's a very good point. RobotFramework has some provisions for [_continue on failure_ behavior](https://robotframework.org/robotframework/latest/RobotFrameworkUserGuide.html#continuing-on-failure), but there doesn't seem to be a global switch we can flip. Sounds like a feature request to me.

File: src/benchmarks/beluga_vs_nav2_multi_dataset/reports/nominal_report/sections/main_corpus.inc (hunk 12 above)
Comment (glpuga) [reply src/benchmarks/beluga_vs_nav2_multi_dataset/reports/nominal_report/sections/main_corpus.inc]: I'm not sure if this is a feature idea or an alternative implementation. 

In any case, as a user, I'd like LAMBKIN to do **less**, not more, in terms of report generation. 

Ideally it should generate data in easy-to-import-elsewhere formats and leave it at that, and probably one easy to  browse format were I can see the results quickly. 

The usual workflow for the engineer will be I run stuff, and I check results, rerun stuff, check results again, rinse and repeat. In this workflow, any time spent getting the pdf file to work around issues in the data is time wasted, and this is probably 95% of the potential users.

If I need to report elsewhere from time to time, most likely the format will not be anything LAMBKIN can produce : web, ieee templated paper for conference, Google docs, email. Most of the time it will also be a one-off, or infrequent enought that I'll not be willing to spend time cohercing lambkin into producing that format directly.

The dream is a csv I can import in a spreadsheet or reprocess with a trashy python script to convert into a latex table that runs as part of the paper make+latex build pipeline.

Comment (hidmic) [reply src/benchmarks/beluga_vs_nav2_multi_dataset/reports/nominal_report/sections/main_corpus.inc]: > I'm not sure if this is a feature idea or an alternative implementation.

Alternative implementation. The current full programmatic approach (as opposed to an interpolated template) is very hard to follow.

> Ideally it should generate data in easy-to-import-elsewhere formats and leave it at that, and probably one easy to browse format were I can see the results quickly.

> The usual workflow for the engineer will be I run stuff, and I check results, rerun stuff, check results again, rinse and repeat. In this workflow, any time spent getting the pdf file to work around issues in the data is time wasted, and this is probably 95% of the potential users.

> If I need to report elsewhere from time to time, most likely the format will not be anything LAMBKIN can produce : web, ieee templated paper for conference, Google docs, email. Most of the time it will also be a one-off, or infrequent enought that I'll not be willing to spend time cohercing lambkin into producing that format directly.

> The dream is a csv I can import in a spreadsheet or reprocess with a trashy python script to convert into a latex table that runs as part of the paper make+latex build pipeline.

Fair point, and you do have that already. Automatic report generation is meant to be a convenience, nevermind the current UX. You can use it but you don't have to use it. You can use [`lambkin.data.access`](https://github.com/Ekumen-OS/lambkin/blob/main/src/core/lambkin-shepherd/src/lambkin/shepherd/data/access.py) APIs on arbitrary benchmark output too, and if you want to use another tool (like a spreadsheet), you can also export the dataframes (once `lambkin` has taken care of aggregating and indexing all the data). It's not well documented though, so there's that.

File: src/benchmarks/beluga_vs_nav2_multi_dataset/Earthfile [hunk 14]
```diff
@@ -0,0 +1,83 @@
...
+    ARG rosdistro  # forward
+    LET user="$(whoami)"
+    LET uid="$(id -u)"
+    LET gid="$(id -g)"
+    BUILD +devel --distro=${distro} --rosdistro=${rosdistro} --user=${user} --uid=${uid} --gid=${gid}
+
+build:
+    ARG distro=jammy
+    ARG rosdistro  # forward
+    FROM lambkin+ubuntu-devel --distro=${distro} --rosdistro=${rosdistro} --components="external/ros2"
```
Comment (hidmic) [inline src/benchmarks/beluga_vs_nav2_multi_dataset/Earthfile:51]: Connected to #114, this should be 
```suggestion
    FROM lambkin+embed-ubuntu-devel --distro=${distro} --rosdistro=${rosdistro} --components="external/ros2"
```
instead.

File: src/benchmarks/beluga_vs_nav2_multi_dataset/reports/nominal_report/sections/datasets/diff_drive_sim_24hs_description.inc [hunk 15]
```diff
@@ -0,0 +1,37 @@
+Dataset information
+^^^^^^^^^^^^^^^^^^^
+
+This is a synthetic dataset created recording a simulated robot wandering around a
+:math:`450 \mathrm{m}^2` office environment for 24 hours.
+
+The simulated diff-drive robot is modelled after a `Kobuki platform <https://iclebo-kobuki.readthedocs.io/en/latest/about.html>`_ featuring an `RPLidar A1 <https://www.slamtec.com/en/Lidar/A1Spec>`_, an 8m range 2D lidar. The
+simulation was performed using Gazebo Classic.
+
+The dataset was recorded in single run, and it consists of a single continuous bagfile with a duration of 24 hours.
```
Comment (hidmic) [inline src/benchmarks/beluga_vs_nav2_multi_dataset/reports/nominal_report/sections/datasets/diff_drive_sim_24hs_description.inc:10]: @glpuga nit:
```suggestion
The dataset was recorded in a single run, and it consists of a single continuous bagfile with a duration of 24 hours.
```

File: src/benchmarks/beluga_vs_nav2_multi_dataset/reports/nominal_report/sections/datasets/diff_drive_sim_24hs_description.inc [hunk 16]
```diff
@@ -0,0 +1,37 @@
...
+During this time the robot is periodically sent navigation goals to reachable locations to ensure it keeps moving
+around the environment.
+
+The robot covers approximately 24 km during the 24 hours, with an average speed of about 0.28 m/s.
```
Comment (hidmic) [inline src/benchmarks/beluga_vs_nav2_multi_dataset/reports/nominal_report/sections/datasets/diff_drive_sim_24hs_description.inc:14]: @glpuga nit: 
```suggestion
The robot covers approximately 24 km during those 24 hours, with an average speed of about 0.28 m/s.
```

File: src/benchmarks/beluga_vs_nav2_multi_dataset/reports/nominal_report/sections/datasets/magazino_datasets_description.inc [hunk 17]
```diff
@@ -0,0 +1,34 @@
...
+which was repurposed to serve as a 2D localization test scenario. The original
+dataset is available `here <https://github.com/magazino/cartographer_magazino>`_.
+
+There's little information about this dataset. In the repository linked
+above it is mentioned that the dataset was recorded
+in a hallway using a Magazino `TORU robot <https://robotsguide.com/robots/toru>`_.
+
+The set contains only two bagfiles, corresponding to two different trajectories along the same hallway.
+"Hallway Localization" is a bit over two minutes long, while "Hallway Return" is a little below four
+minutes long.
```
Comment (hidmic) [inline src/benchmarks/beluga_vs_nav2_multi_dataset/reports/nominal_report/sections/datasets/magazino_datasets_description.inc:14]: @glpuga nit: shall we use numbers to talk about duration for consistency? Or maybe use both letters and numbers?

File: src/benchmarks/beluga_vs_nav2_multi_dataset/reports/nominal_report/sections/datasets/omni_drive_sim_24hs_description.inc [hunk 18]
```diff
@@ -0,0 +1,40 @@
...
+`AWS Robomaker Bookstore World <https://github.com/aws-robotics/aws-robomaker-bookstore-world>`_ for 24 hours.
+
+The simulated omni-drive robot is modelled after a customized `Robomaster EP <https://www.dji.com/global/robomaster-ep>`_ featuring an `RPLidar A2 <https://www.slamtec.com/en/Lidar/A2Spec>`_, a 12m range 2D lidar.
+The simulation was performed using Gazebo Sim.
+
+The dataset was recorded in a single run, and it consists of a single continuous bagfile with a duration of 24 hours.
+During this time the robot is periodically sent navigation goals to reachable locations within the map to ensure it
+keeps moving.
+
+The robot covers approximately 45 km during the 24 hours, with an average speed of about 0.5 m/s.
```
Comment (hidmic) [inline src/benchmarks/beluga_vs_nav2_multi_dataset/reports/nominal_report/sections/datasets/omni_drive_sim_24hs_description.inc:14]: @glpuga nit: 
```suggestion
The robot covers approximately 45 km during those 24 hours, with an average speed of about 0.5 m/s.
```

File: src/benchmarks/beluga_vs_nav2_multi_dataset/reports/nominal_report/sections/datasets/willow_garage_description.inc [hunk 19]
```diff
@@ -0,0 +1,56 @@
...
+the original dataset as distributed in here:
+
+- The original ROS 1 bagfiles were converted to ROS 2 bagfiles.
+- All topics not related to lidar-based 2D localization were removed to reduce file size.
+- The map->odom_combined TF transformations were removed from the transform tree.
+- The "/" prefix in all transform frame names was removed to avoid issues with the ROS 2 TF tree.
+- A reference ground-truth and a matching occupancy map were generated for each bagfile using SLAM.
+
```
Comment (hidmic) [inline src/benchmarks/beluga_vs_nav2_multi_dataset/reports/nominal_report/sections/datasets/willow_garage_description.inc:47]: @glpuga nit: remove trailing blank lines

File: src/benchmarks/beluga_vs_nav2_multi_dataset/reports/nominal_report/sections/main_corpus.inc [hunk 20]
```diff
@@ -0,0 +1,245 @@
+
+.. repl-quiet::
+
+    def get_bagfile_info(bagpath):
```
Comment (hidmic) [inline src/benchmarks/beluga_vs_nav2_multi_dataset/reports/nominal_report/sections/main_corpus.inc:4]: @glpuga fyi, feel free to ignore: there's also [`rosbag2_py.Info().read_metadata(bag_path)`](https://github.com/ros2/rosbag2/blob/786c3c4b8ab05271630371cf515130ba02a9cde8/rosbag2_py/rosbag2_py/_info.pyi#L9) for this too.

File: src/benchmarks/beluga_vs_nav2_multi_dataset/reports/nominal_report/sections/platform.inc [hunk 21]
```diff
@@ -0,0 +1,131 @@
...
+   * - `Cartographer Magazino <https://github.com/magazino/cartographer_magazino>`_
+     - Two sample datasets for mapping and localization that were recorded in a hallway.
+   * - `OpenLORIS-Scene dataset (Office set) <https://lifelong-robotic-vision.github.io/dataset/scene.html>`_
+     - Short trajectories in an office environment.
+
+None of these datasets were originally conceived for the evaluation of 2D localization systems,
+and therefore none of them provided a complete ground-truth that included both the real-world trajectory and the environment's
+occupancy grid map.
+
+For this reason ground-truth data was generated locally from the bagfiles themselves using Cartographer SLAM to create mutually
```
Comment (hidmic) [inline src/benchmarks/beluga_vs_nav2_multi_dataset/reports/nominal_report/sections/platform.inc:33]: @glpuga nit: 
```suggestion
For this reason, ground-truth data was generated locally from the bagfiles themselves using Cartographer SLAM to create mutually
```

File: src/benchmarks/beluga_vs_nav2_multi_dataset/reports/nominal_report/sections/platform.inc [hunk 22]
```diff
@@ -0,0 +1,131 @@
...
+consistent ground-truth trajectory and occupancy grid map for each bagfile.
+
+It has to be notices that this approach has some drawbacks that need to be taken into account when interpreting the results later:
```
Comment (hidmic) [inline src/benchmarks/beluga_vs_nav2_multi_dataset/reports/nominal_report/sections/platform.inc:36]: @glpuga nit: 
```suggestion
Note that this approach has some drawbacks that need to be taken into account when interpreting the results later:
```

File: src/benchmarks/beluga_vs_nav2_multi_dataset/reports/nominal_report/sections/platform.inc [hunk 23]
```diff
@@ -0,0 +1,131 @@
...
+
+- Both the space and time diversity of the original datasets is lost, as the ground-truth is generated for each individual bagfile representing the state of the environment as perceived when the data was recorded.
+- All of the objects present in the environment are part of the generated occupancy grid map; i.e. there are no unmapped objects.
+
+These real robot datasets have two more limitations worth mentioning:
```
Comment (hidmic) [inline src/benchmarks/beluga_vs_nav2_multi_dataset/reports/nominal_report/sections/platform.inc:41]: @glpuga nit: 
```suggestion
These real-world datasets have two more limitations worth mentioning:
```
or use `real robot` instead of `real world` everywhere else, for consistency.

File: src/benchmarks/beluga_vs_nav2_multi_dataset/reports/nominal_report/sections/platform.inc [hunk 24]
```diff
@@ -0,0 +1,131 @@
...
+
+- The longest run time of any bagfile the real robot datasets listed above is around 30 minutes, which may not detect longer-term resource use issues, such as memory leaks.
```
Comment (hidmic) [inline src/benchmarks/beluga_vs_nav2_multi_dataset/reports/nominal_report/sections/platform.inc:43]: @glpuga nit: 
```suggestion
- The longest run-time of any of the real-world datasets listed above is around 30 minutes, which may fail to flag longer-term resource usage issues, such as memory leaks.
```

**Review by hidmic — APPROVED** [review]

File: src/benchmarks/beluga_vs_nav2_multi_dataset/reports/nominal_report/sections/datasets/willow_garage_description.inc (hunk 19 above)
Comment (glpuga) [reply src/benchmarks/beluga_vs_nav2_multi_dataset/reports/nominal_report/sections/datasets/willow_garage_description.inc:47]: Fixed.

File: src/benchmarks/beluga_vs_nav2_multi_dataset/reports/nominal_report/sections/datasets/magazino_datasets_description.inc (hunk 17 above)
Comment (glpuga) [reply src/benchmarks/beluga_vs_nav2_multi_dataset/reports/nominal_report/sections/datasets/magazino_datasets_description.inc:14]: Numbers sounds good.

File: src/benchmarks/beluga_vs_nav2_multi_dataset/reports/nominal_report/sections/main_corpus.inc (hunk 20 above)
Comment (glpuga) [reply src/benchmarks/beluga_vs_nav2_multi_dataset/reports/nominal_report/sections/main_corpus.inc:4]: Lets keep it in mind for the next time, but since this has no other testing than surving running report generation with the playground in a bunch of different error states, let's keep it like this. I'll add a TODO so that we update this when we update  a do a new measurement campaign.

**Review by hidmic — APPROVED** [review]
